### PR TITLE
Fix auto-scrolling and add option to download or tail big files

### PR DESCRIPTION
**Pull request by iverberk**
This commit makes sure that the browser is not overwhelmed with logs
from big files. It auto-scrolls by default but if you manually move the
scrollbar it will stay fixed in that position.

This is needs to be tested properly before merging as there are some rough edges. I'll create file downloading in another PR.

### CODE REVIEW

**Comment by jippi** [comment]
LGTM

Would be nice in further PR til either strip shell colors codes, or convert them into their html versions :

**Comment by iverberk** [comment]
This is still WIP, will be done tomorrow I think. I'm going to be a little a smarter about when to show the whole file and when to tail the last lines. You can download files now too in case of binary or too large.

File: backend/connection.go [hunk 1]
```diff
@@ -18,9 +18,12 @@ const (
 	// many lines to tail from.
 	bytesToLines int64 = 120
 
-	// defaultTailLines is the number of lines to tail by default if the value
-	// is not overriden.
-	defaultTailLines int64 = 30
+	// defaultTailLines is the number of lines to tail by default.
+	defaultTailLines int64 = 1000
```
Comment (jippi) [inline backend/connection.go]: thats a lot of lines!

File: frontend/src/components/allocation/files.js [hunk 2]
```diff
@@ -170,12 +171,27 @@ class AllocationFiles extends Component {
...
         }
 
+        const oversizedWarning = !this.props.file.Oversized ? '' :
+          <span>
+            <i className="pe-7s-attention" data-tip data-for={ `tooltip-${this.props.file.File}` }></i>
+            <span>
+              <ReactTooltip id={ `tooltip-${this.props.file.File}` }>
+                <span className="file-size-warning">
+                  The file you are trying to view is too large.<br />
+                  Tailing has started from the last thousand lines. <br />
```
Comment (jippi) [inline frontend/src/components/allocation/files.js]: i would put `1.000 lines` instead of words :)

File: backend/connection.go (hunk 1 above)
Comment (iverberk) [inline backend/connection.go]: yeah, too much? I don't think it will overwhelm the browser and this way you do get a nice piece of history up-front.

**Comment by jippi** [comment]
very much like supervisord, it would be nice to be able to tail a file in full-screen width.. basically a new tab with no design (left/top menu), but just white background and stream of content tailing out like `tail -f`  - basically download without content disposition and a constant stream instead

File: backend/connection.go (hunk 1 above)
Comment (jippi) [inline backend/connection.go]: would have to test it out in production, for me, usually a few 100 lines are way more than enough. 1000 lines is overwhelming :)

**Comment by iverberk** [comment]
With the new tailing functionality we could easily do this. Would require a no-design screen/route for which we pass the necessary information to start tailing. Let's create a separate PR for this. This will help most people right away I think.

**Comment by jippi** [comment]
ship it :)

**Pull request merged**